### PR TITLE
Add create S3 Bucket panel.

### PR DESCRIPTION
**Pull request by zhangzhx**
Fix the S3 bucket creation dialog doesn't show error message

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Related Issue(s)
#273 

## Screenshots (if appropriate)

s3bucketcreation

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialogTest.kt [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+class CreateS3BucketDialogTest {
+    @Rule
+    @JvmField
+    val projectRule = ProjectRule()
+
+    private val s3Mock = mock<S3Client>(defaultAnswer = DelegateSdkConsumers())
+
+    @Before
+    fun setUp() {
+        reset(s3Mock)
```
Comment (richa-li) [inline jetbrains-core/tst/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialogTest.kt]: Not sure if this is required, I thought Mockito/JUnit reset the class class for each test method

Comment (zhangzhx) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialogTest.kt]: Nice catch. Removed.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    private fun bucketName(): String = view.bucketName.text.trim()
+
+    private fun createBucket() {
+        s3Client.createBucket { request -> request.bucket(bucketName()) }
+        bucket = bucketName()
+    }
+
+    @TestOnly
+    fun validateBucketName(): String? = if (bucketName().isEmpty()) message("s3.create.bucket.missing.bucket.name") else null
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt]: Should we do more validation - s3 publishes a pretty good set of things that it'll accept.

ps - why expose this method? can't we just call `doValidate` directly?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+class CreateS3BucketDialog(
+    private val project: Project,
+    private val s3Client: S3Client,
+    private val parent: Component? = null
+) : DialogWrapper(project, parent, false, IdeModalityType.PROJECT) {
+
+    private val view = CreateBucketPanel(project)
+
+    var bucket: String? = null
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt]: Why do we need the `bucket` instance variable, the `bucketName()` method and the access from the testfield ?

Reckon we can just fetch it straight from the text field.... no need to have two mutable states.

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialogTest.kt (hunk 1 above)
Comment (kiiadi) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialogTest.kt]: The reset will only happen if you're using the `@RunWith(JUnitMockitoRunner)` which we ain't.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt (hunk 2 above)
Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt]: S3 bucket naming has many restrictions and they are changing overtime. https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html

The SDKs don't do client side validations and I have the same concerns. Ie. if S3 removes some restriction, the old version of the plugin won't allow customer to create bucket.

Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt]: `doValidate` is a `protected` method.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt (hunk 3 above)
Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt]: I was thinking that `bucketName` always returns whatever is in the text field whether is validated or not. `bucket` returns the validated value which is safe to use outside of the Panel.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateBucketPanel.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import javax.swing.JTextField;
+
+public class CreateBucketPanel {
+    private final Project project;
+
+    JTextField bucketName;
+    JPanel component;
+
+    public CreateBucketPanel(Project project) {
+        this.project = project;
```
[the remark this answers is not in the document]
Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateBucketPanel.java]: Removed this.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateBucketPanel.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.intellij.openapi.project.Project;
+
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+
+public class CreateBucketPanel {
+    private final Project project;
+
+    JTextField bucketName;
```
[the remark this answers is not in the document]
Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateBucketPanel.java]: Added.

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt (hunk 3 above)
Comment (zhangzhx) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/CreateS3BucketDialog.kt]: Removed the `bucket` field as it doesn't gain too much. Using `bucketName()` instead

**Pull request merged**